src/tasks/DockerComposeTaskDefinitionBase.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { TaskDefinitionBase } from './TaskDefinitionBase';
+
+export interface DockerComposeUpOptions {
+    detached?: boolean;
+    build?: boolean;
+    scale?: { [service: string]: number };
+    services?: string[];
+    customOptions?: string;
+}
+
+export interface DockerComposeDownOptions {
+    removeImages?: 'all' | 'local';
+    removeVolumes?: boolean;
+    customOptions?: string;
+}
+
+export interface DockerComposeOptions {

```

Comment:
I imagine there's a way in TypeScript to describe the fact that `up` or `down` should be included, but not both, and not neither.

Reply:
Looks like it, but it involves creating intermediate types where one is `never` and the other is defined; and then a union of those two is the "final" type: https://stackoverflow.com/a/61281828

I think it's sort of an awkward syntax...

Reply:
I found what I think is a reasonable way to do it. It doesn't offer a whole lot, only prevents you from doing this:

```typescript
const foo: DockerComposeOptions = {
    up: { ... },
    down: { ... },
    files: [ ... ],
};
```

In the above example, it forces you to define either `up` or `down` but not both. Having neither is allowed by this syntax though. I guess it's still an improvement, plus I learned something new about TypeScript!